src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2695,8 +2695,8 @@ void smt2_convt::flatten_array(const exprt &expr)
   PRECONDITION(size_expr.id() == ID_constant);
 
   mp_integer size = numeric_cast_v<mp_integer>(to_constant_expr(size_expr));
-  CHECK_RETURN_WITH_DIAGNOSTICS(size != 0, "can't convert zero-sized array");
-
+  if(size == 0)
+    return;
```

Comment:
This is suspicious -- this produces an SMT file that might now parse.
Did you want to make that a precondition, and then check at the call site?

Reply:
I reverted here to the original `CHECK_RETURN` and pushed the check to the above `convert_struct`.